app/src/app/voting/voting.component.ts

```diff
@@ -48,7 +52,12 @@ export class VotingComponent implements OnInit {
           })
         })
     )
-
+    this.db.object("landsraad/votingConfig/resultsShown").valueChanges().subscribe((value: boolean) => {
+      this.showResults = value
+      if (this.showResults) {
+        this.composeAndSetData()
```

Comment:
The correct way would be that all data for the table will be Observable. You initialize observable in `ngOnInit()`. When the `showResults` flag is switched, the UI will subscribe to the data (with `| async`) and it will be fetched. But this way also works, give it half an hour and if it's too complex refactoring, we can keep it like this.

Reply:
Ok I tried to refactor, but I don't know enough about rxjs to refactor successfully. So maybe we can keep as is.